openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala

```diff
@@ -127,12 +134,103 @@ trait JsonSchemas extends algebra.JsonSchemas with TuplesSchemas {
 
     }
 
-    case class Field(
-        name: String,
-        tpe: DocumentedJsonSchema,
-        isOptional: Boolean,
-        documentation: Option[String]
-    )
+    final class Field private (
+        val name: String,
+        val tpe: DocumentedJsonSchema,
+        val isOptional: Boolean,
+        val defaultValue: Option[ujson.Value],
+        val documentation: Option[String]
+    ) extends Serializable
+        with Product
+        with Equals {
+
+      @deprecated("`Field` is no longer a `case class` and won't implement `Equals`", "3.1.0")
+      override def canEqual(other: Any): Boolean = other.isInstanceOf[Field]
+      @deprecated("`Field` is no longer a `case class` and won't implement `Product`", "3.1.0")
+      override def productArity: Int = 5
+      @deprecated("`Field` is no longer a `case class` and won't implement `Product`", "3.1.0")
+      override def productElement(idx: Int): Any = idx match {
+        case 0 => name
+        case 1 => tpe
+        case 2 => isOptional
+        case 3 => defaultValue
+        case 4 => documentation
+        case _ => throw new IndexOutOfBoundsException(idx.toString)
+      }
+      override def toString: String = s"Field($name,$tpe,$isOptional,$defaultValue,$documentation)"
+      override def hashCode: Int = Hashing.hash(name, tpe, isOptional, defaultValue, documentation)
+      override def equals(other: Any): Boolean = other match {
+        case field: Field =>
+          name == field.name && tpe == field.tpe && isOptional == field.isOptional &&
+            defaultValue == field.defaultValue && documentation == field.documentation
+        case _ => false
```

Comment:
@julienrf do you know the "right" way to handle this Scala issue (IMHO, really a wart)? 

```
[warn] openapi/openapi/src/main/scala/endpoints4s/openapi/JsonSchemas.scala:163:14: The outer reference in this type test cannot be checked at run time.
[warn]         case field: Field =>
[warn]              ^
[warn] one warning found
```

Should we switch `Field` back to not be `final`?

Reply:
Good point, maybe we can use `deprecatedInheritance` instead?